File: front-end/not-your-simple-todo/src/services/WorkspaceService.js
import axiosInstance from './axiosInstance';

const WorkspaceService = {
  getWorkspacesOfUser: async () => {
    const response = await axiosInstance.get('workspace/user/');
    return response.data;
  },

  getWorkspace: async (id) => {
    const response = await axiosInstance.get(`workspace/${id}`);
    return response.data;
  },

  createWorkspace: async (workspace) => {
    const response = await axiosInstance.post('workspace/', workspace);
    return response.data;
  },

  updateWorkspace: async (workspace) => {
    const response = await axiosInstance.put(`workspace/`, workspace);
    return response.data;
  },

  deleteWorkspace: async (id) => {
    const response = await axiosInstance.delete(`workspace/${id}`);
    return response.data;
  },

  // /{workspace-id}/add-todo/{id}
  addTodo: async (workspaceId, todoId) => {
    const response = await axiosInstance.put(`workspace/${workspaceId}/add-todo/${todoId}`);
    return response.data;
  },
  addUserToWorkspace: async(workspaceId, userIds) => {
    const response = await axiosInstance.put(`workspace/user/${workspaceId}`, userIds);
    return response.data;
  }
};

export default WorkspaceService;

File: not-your-simple-todo/src/main/java/net/lakhwan/not_your_simple_todo/service/WorkspaceService.java
package net.lakhwan.not_your_simple_todo.service;

import net.lakhwan.not_your_simple_todo.dto.Todo.TodoDto;
import net.lakhwan.not_your_simple_todo.dto.WorkspaceDto;

import java.util.List;
import java.util.Set;

public interface WorkspaceService {

    WorkspaceDto createWorkspace(WorkspaceDto workspaceDto, String username);

    List<WorkspaceDto> getAllWorkspaces ();

    Set<WorkspaceDto> getWorkspacesOfUser(Long userId);

    Set<TodoDto> getTodosOfWorkspace(Long workspaceId);

    WorkspaceDto updateAssignUsersToWorkspace(Long workspaceId, Set<Long> userIds);

    WorkspaceDto updateWorkspaceName(WorkspaceDto updatedWorkspaceDto);

    void deleteWorkspace(Long workspaceId);

    WorkspaceDto addTodo(Long workspaceId, Long todoId);
}
